pkg/pgxconn/pgx_conn.go

```diff
@@ -0,0 +1,67 @@
+package pgxconn
```

Comment:
sorry, but why do we require a new package that just replicates the pgconn? This leads to the entire changes across the repository, increasing the functional stack call which adds to performance (since we call some functions in millions).

Reply:
If any strong reason, then maybe we should add that explanation here as comments.

Reply:
we need the PgxConn interface to write unit tests, pgxpool.Pool is a struct and we can't mock it

Reply:
and since all of these functions are one liners they will most likely be inlined on compile time